aquadoggo/src/db/stores/document.rs

```diff
@@ -0,0 +1,296 @@
+            ",
+        )
+        .bind(document_view.id().as_str())
+        .bind(schema_id.as_str())
+        .execute(&self.pool)
+        .await
+        .map_err(|e| DocumentStorageError::FatalStorageError(e.to_string()))?;
+
+        // Check every insertion performed affected exactly 1 row.
+        if document_view_insertion_result.rows_affected() != 1
+            || field_relations_insertion_result
+                .iter()
+                .any(|query_result| query_result.rows_affected() != 1)
+        {
+            return Err(DocumentStorageError::InsertionError(
+                document_view.id().clone(),
+            ));
+        }
+
+        // Commit the transaction.
+        transaction
+            .commit()
+            .await
+            .map_err(|e| DocumentStorageError::FatalStorageError(e.to_string()))?;
+
+        Ok(())
+    }
+
+    /// Get a document view from the database by it's id.
+    async fn get_document_view_by_id(
+        &self,
+        id: &DocumentViewId,
+    ) -> Result<StorageDocumentView, DocumentStorageError> {
+        let document_view_field_rows = query_as::<_, DocumentViewFieldRow>(
+            "
+            SELECT
+                document_view_fields.document_view_id,
+                document_view_fields.operation_id,
+                document_view_fields.name,
+                operation_fields_v1.field_type,
+                operation_fields_v1.value
+            FROM
+                document_view_fields
+            LEFT JOIN operation_fields_v1
+                ON
+                    operation_fields_v1.operation_id = document_view_fields.operation_id
+                AND
+                    operation_fields_v1.name = document_view_fields.name
+            WHERE
+                document_view_fields.document_view_id = $1
+            ",
+        )
+        .bind(id.as_str())
+        .fetch_all(&self.pool)
+        .await
+        .map_err(|e| DocumentStorageError::FatalStorageError(e.to_string()))?;
+
+        Ok(StorageDocumentView::new(
+            &id.to_owned(),
+            &parse_document_view_field_rows(document_view_field_rows),
+        ))
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use std::convert::TryFrom;
+    use std::str::FromStr;
+
+    use p2panda_rs::document::{DocumentViewFields, DocumentViewId, DocumentViewValue};
+    use p2panda_rs::entry::{LogId, SeqNum};
+    use p2panda_rs::identity::{Author, KeyPair};
+    use p2panda_rs::operation::{AsOperation, OperationId};
+    use p2panda_rs::schema::SchemaId;
+    use p2panda_rs::storage_provider::traits::{AsStorageEntry, EntryStore};
+    use p2panda_rs::test_utils::constants::{DEFAULT_HASH, DEFAULT_PRIVATE_KEY, TEST_SCHEMA_ID};
+
+    use crate::db::models::entry::EntryRow;
+    use crate::db::stores::document::{DocumentStore, StorageDocumentView};
+    use crate::db::stores::test_utils::{test_create_operation, test_db};
+
+    fn entries_to_document_views(entries: &Vec<EntryRow>) -> Vec<StorageDocumentView> {
```

Comment:
I think i should be using the new mocking features on the stores for these tests...